src/clarity-angular/datagrid/render/main-renderer.ts

```diff
@@ -47,7 +49,22 @@ export class DatagridMainRenderer implements AfterContentInit, AfterViewInit, On
      * Makes each header compute its width.
      */
     private computeHeadersWidth() {
+
+        let nbColumns: number = this.headers.length;
+        let nbStrictWidthColumns: number = this.headers.toArray().filter(column => column.strictWidth).length;
```

Comment:
You're using the old width array, since you're about to compute it. So I'm expecting bugs here. You should just be using a boolean that becomes true if you see at least one non-strict column as you iterate.